client/extensions/woocommerce/state/sites/customers/selectors.js

```diff
@@ -1,86 +1,59 @@
-	return false === inFlight;
-};
+	return inFlight( state, siteId, searchTerm ) === false;
+}
 
 /**
  * @param {object} state Whole Redux state tree
  * @param {string} searchTerm Search term to check
  * @param {number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @returns {boolean} Whether the search results for a given term are currently being retrieved from the server.
  */
-export const isCustomerSearchLoading = (
-	state,
-	searchTerm,
-	siteId = getSelectedSiteId( state )
-) => {
-	const inFlight = get(
-		state,
-		`extensions.woocommerce.sites[${ siteId }].customers.isSearching[${ searchTerm }]`
-	);
+export function isCustomerSearchLoading( state, searchTerm, siteId = getSelectedSiteId( state ) ) {
 	// Strict check because it could also be undefined.
-	return true === inFlight;
-};
+	return inFlight( state, siteId, searchTerm ) === true;
+}
 
 /**
  * @param {object} state Whole Redux state tree
  * @param {string} searchTerm Search term to check
  * @param {number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @returns {Array} List of customer results matching term
  */
-export const getCustomerSearchResults = (
-	state,
-	searchTerm,
-	siteId = getSelectedSiteId( state )
-) => {
+export function getCustomerSearchResults( state, searchTerm, siteId = getSelectedSiteId( state ) ) {
 	if ( ! isCustomerSearchLoaded( state, searchTerm, siteId ) ) {
 		return [];
 	}
 
-	const customers = get( state, `extensions.woocommerce.sites[${ siteId }].customers.items`, {} );
-	const customerIdsForTerm = get(
-		state,
-		`extensions.woocommerce.sites[${ siteId }].customers.queries[${ searchTerm }]`,
-		[]
-	);
-	return filter( customerIdsForTerm.map( id => customers[ id ] || false ) );
-};
+	const customers = state?.extensions?.woocommerce?.sites?.[ siteId ]?.customers?.items ?? {};
+	const customerIdsForTerm =
+		state?.extensions?.woocommerce?.sites?.[ siteId ]?.customers?.queries?.[ searchTerm ] ?? [];
+	return customerIdsForTerm.map( id => customers[ id ] || false ).filter( customer => !! customer );
+}
 
 /**
  * @param {object} state Whole Redux state tree
  * @param {number} customerId Customer ID to get
  * @param {number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
- * @returns {object|False} a customer object as stored in the API, false if not found
+ * @returns {object|boolean} a customer object as stored in the API, false if not found
  */
-export const getCustomer = ( state, customerId, siteId = getSelectedSiteId( state ) ) => {
-	return get(
-		state,
-		`extensions.woocommerce.sites[${ siteId }].customers.items[${ customerId }]`,
-		false
-	);
-};
+export function getCustomer( state, customerId, siteId = getSelectedSiteId( state ) ) {
+	const customer =
+		state?.extensions?.woocommerce?.sites?.[ siteId ]?.customers?.items?.[ customerId ];
+	return customer !== undefined ? customer : false;
```

Comment:
Note that this could be simplified further, to `customer ?? false`, but that would behave differently for the `null` value, where it's now being returned as `null` and would in that case also be returned as `false`.

I opted to preserve existing behaviour here and in some other places, but if you think it's safe to simplify further and would like me to, I'd be more than happy to make the change.

Reply:
It makes sense to keep what is already there in terms of behavior. I wonder if we could use the opportunity to fix it, though - judging by the jsdoc, this function is not intended to return a `null` value and actually should return `false` in that case.